CONTRIBUTION.md

```diff
@@ -0,0 +1,245 @@
+  Mac OS X?  (10.7.4? 10.9.0?) Linux? (Which distro? Which version of that
+  distro? 32 or 64 bits?) Again, more detail is better.
+* **Which version or versions of the software are you using?** Ideally, you
+  followed the advice above and have ruled out (or verified that the problem
+  exists in) a few different versions.
+* **How can the developers recreate the bug on their end?** If possible,
+  include a copy of your code, the command you used to invoke it, and the full
+  output of your run (if applicable.)
+
+    * A common tactic is to pare down your code until a simple (but still
+      bug-causing) "base case" remains. Not only can this help you identify
+      problems which aren't real bugs, but it means the developer can get to
+      fixing the bug faster.
+
+
+Contributing changes
+====================
+
+Licensing of contributed material
+---------------------------------
+
+Keep in mind as you contribute, that code, docs and other material submitted to
+open source projects are usually considered licensed under the same terms
+as the rest of the work.
+
+The details vary from project to project, but from the perspective of this
+document's authors:
+
+- Anything submitted to a project falls under the licensing terms in the
+  repository's top level ``LICENSE`` file.
+
+    - For example, if a project's ``LICENSE`` is BSD-based, contributors should
+      be comfortable with their work potentially being distributed in binary
+      form without the original source code.
+
+- Per-file copyright/license headers are typically extraneous and undesirable.
+  Please don't add your own copyright headers to new files unless the project's
+  license actually requires them!
+
+    - Not least because even a new file created by one individual (who often
+      feels compelled to put their personal copyright notice at the top) will
+      inherently end up contributed to by dozens of others over time, making a
+      per-file header outdated/misleading.
+
+Version control branching
+-------------------------
+
+* Always **make a new branch** for your work, no matter how small. This makes
+  it easy for others to take just that one set of changes from your repository,
+  in case you have multiple unrelated changes floating around.
+
+    * A corollary: **don't submit unrelated changes in the same branch/pull
+      request**! The maintainer shouldn't have to reject your awesome bugfix
+      because the feature you put in with it needs more review.
+
+* **Base your new branch off of the appropriate branch** on the main
+  repository:
+
+    * **Bug fixes** should be based on the branch named after the **oldest
+      supported release line** the bug affects.
+
+        * E.g. if a feature was introduced in 1.1, the latest release line is
+          1.3, and a bug is found in that feature - make your branch based on
+          1.1.  The maintainer will then forward-port it to 1.3 and master.
```

Comment:
this section needs to be reworked according to how we want to operate. At the moment we've only used the master branch, and create releases off of what has changed since then. We should decide on a clear plan of how we want to control features and bugfixes going into a release.